### DISCUSSION TITLE
Concurrent Firebase.initializeApp() Calls during App Initialization

### DISCUSSION THREAD

**Discussion by dusanristic** [invertase/react-native-firebase#7611] [Q&A]
I have a following question:

What will happen if I call `firebase.initializeApp(config, name)` from multiple places simultaneously? Can a crash occur due to a race condition, and possible multiple instances creation, or is there some queue that will handle this kind of situation? Just to be clear, I am not talking about calling initializeApp after the app was initialized; I am talking about calling `firebase.initializeApp` while the app initialization is in progress.

Is there any built-in mechanism to handle this specific situation, or is it up to us to take care of this?

**Comment by mikehardy** [comment]
Unknown. I would avoid it by design if possible. I believe the underlying SDKs guard against it but can't be certain, never had issue reports related to it either way
